src/AndersonAcceleration.f90

```diff
@@ -72,183 +71,288 @@ MODULE AndersonAccelerationTypes
+      tmpA=BLAS_dot(solver%tmpvec,solver%tmpvec)
+      tmpb=BLAS_dot(solver%r(2),solver%tmpvec)
+      solver%alpha(1)=tmpb/tmpA
+    ELSEIF(depth_s > 1) THEN
+      !Construct coefficient matrix, right hand side, and solve for fit coefficients
+      DO i=1,depth_s
+        CALL BLAS_copy(solver%r(i),x_new)
+        CALL BLAS_axpy(solver%r(depth_s+1),x_new,-1.0_SRK)
+        tmpb=BLAS_dot(solver%r(depth_s+1),x_new)
+        CALL solver%LS%b%set(i,tmpb)
+        DO k=1,i
+          CALL BLAS_copy(solver%r(depth_s+1),solver%tmpvec)
+          CALL BLAS_axpy(solver%r(k),solver%tmpvec,-1.0_SRK)
+          tmpA=BLAS_dot(solver%tmpvec,x_new)
+          CALL solver%LS%A%set(k,i,tmpA)
+        ENDDO
+      ENDDO
+      CALL solver%LS%solve()
+      DO i=1,depth_s
+        CALL solver%LS%x%get(i,solver%alpha(i))
+      ENDDO
+    ENDIF
+
+    !Back out "0th" alpha
+    solver%alpha(depth_s+1)=1.0_SRK
+    DO i=1,depth_s
+      solver%alpha(depth_s+1)=solver%alpha(depth_s+1)-solver%alpha(i)
+    ENDDO
+
+    !Ensure Anderson coefficient solve did not fail due to bad input vector
+    IF(ISNAN(solver%alpha(depth_s+1)) .OR. (solver%alpha(1) .APPROXEQ. 0.0_SRK)) THEN
+      !Bad Anderson coefficient solve, revert to under-relaxation and reset Anderson solver
+      CALL x_new%set(0.0_SRK)
+      CALL BLAS_axpy(solver%Gx(1),x_new,solver%beta)
+      CALL BLAS_axpy(solver%x(1),x_new,1.0_SRK-solver%beta)
+      CALL solver%reset(x_new)
+    ELSE
+      !Get accelerated solution
+      CALL x_new%set(0.0_SRK)
+      DO i=1,depth_s+1
+        CALL BLAS_copy(solver%x(i),solver%tmpvec)
+        CALL BLAS_axpy(solver%r(i),solver%tmpvec,solver%beta)
+        CALL BLAS_axpy(solver%tmpvec,x_new,solver%alpha(i))
+      ENDDO
+
+      !Push back solution vectors and load in newest accelerated as next initial iterate
+      DO i=MIN(depth_s+1,solver%depth),1,-1
+        CALL BLAS_copy(solver%x(i),solver%x(i+1))
+      ENDDO
+      CALL BLAS_copy(x_new,solver%x(1))
+    ENDIF
   ELSE
-    CALL Anderson_Update(solver%id)
+    !Get under-relaxed solution using mixing parameter
+    CALL BLAS_scal(x_new,solver%beta)
+    CALL BLAS_axpy(solver%x(1),x_new,1.0_SRK-solver%beta)
+    CALL BLAS_copy(x_new,solver%x(1))
   ENDIF
 
-  solver%iter=solver%iter+1
-#endif
 ENDSUBROUTINE step_AndersonAccelerationType
 !
 !-------------------------------------------------------------------------------
-!> @brief Reset the Anderson acceleration solver
-!> @param solver The anderson solver to act on
-!>
-!> This routine resets the counters for the anderson acceleration
+!> @brief Set or reset the initial iterate and linear solver state for the Anderson solver
+!> @param solver The Anderson solver to act on
+!> @param x Initial iterate solve is starting from
 !>
-SUBROUTINE reset_AndersonAccelerationType(solver)
+SUBROUTINE reset_AndersonAccelerationType(solver,x)
+  CHARACTER(LEN=*),PARAMETER :: myName='reset_AndersonAccelerationType'
```

Comment:
slide this down below the dummy arguments

Reply:
Are we changing our format to do it that way?

Reply:
We've pretty much always done it that way...  Most often you see this:

```
SUBROUTINE stuff(arg1,arg2)
  TYPE(*) :: arg1
  TYPE(*) :: arg2
  !
  CHARACTER(LEN=*),PARAMETER :: myName='stuff'

  !Rest of subroutine
ENDSUBROUTINE stuff
```

Reply:
We must be looking at different parts of the code entirely. I almost can't find an example of this in Futility or MPACT.

Reply:
Well, that's how the standard is written.  Much of MPACT follows this.  Futility is probably much more a mixed bag since it's so old.  But yes, the rule is that the the dummy arguments each be on their own line immediately following the function/subroutine declaration line.  Other variables come after.

Reply:
Okay, taken care of.